cache/redis/redis.go

```diff
@@ -27,10 +27,13 @@ func New(ctx context.Context, opt Options) (*Cache, error) {
 // Get executes a lookup and returns whether a key exists in the cache along with its value.
 func (c *Cache) Get(key string) (interface{}, bool, error) {
 	res, err := c.rdb.Do(c.ctx, "get", key).Result()
-	if err == redis.Nil || err != nil {
-		return nil, false, err
+	if err == nil {
```

Comment:
I would suggest to use the usual pattern and check first for the err != nil. Inside you check specifically for redis.Nil.
the happy path should be the last statement...